test/parameters.jl

```diff
@@ -49,6 +49,20 @@ pdiagmat(args...) = PDiagMat(args...)
         )
     end
 
+    @testset "orthogonal" begin
+        is_almost_orthogonal(X::AbstractMatrix, tol) = norm(X'X - I) < tol
+        @testset "nearest_orthogonal_matrix" begin
+            X_orth = ParameterHandling.nearest_orthogonal_matrix(randn(5, 4))
+            @test is_almost_orthogonal(X_orth, 1e-9)
+            X_orth_2 = ParameterHandling.nearest_orthogonal_matrix(X_orth)
+            @test X_orth ≈ X_orth_2 # nearest_orthogonal_matrix is a projection.
+        end
+
+        X = orthogonal(randn(5, 4))
+        @test X == X
```

Comment:
maybe also test `@test is_almost_orthogonal(value(X), 1e-9)`? Do we need to test its compatibility with AD?

Reply:
Good call, will do.